public/containers/ClearCacheModal/ClearCacheModal.tsx

```diff
@@ -0,0 +1,233 @@
+      INDEX_OP_BLOCKS_TYPE.META_DATA,
+      INDEX_OP_BLOCKS_TYPE.READ_ONLY,
+      INDEX_OP_BLOCKS_TYPE.READ_ONLY_ALLOW_DELETE,
+    ];
+    if (!!services && visible) {
+      setLoading(true);
+      switch (type) {
+        case INDEX_OP_TARGET_TYPE.DATA_STREAM:
+          setHint("Cache will be cleared for the following data streams.");
+          setBlockHint(
+            "Cache will not be cleared for the following data streams because one or more backing indexes are closed or blocked."
+          );
+          filterBlockedItems(services, selectedItems, indexBlocksTypes, INDEX_OP_TARGET_TYPE.DATA_STREAM, false)
+            .then((filteredDataStreamsResult) => {
+              setUnBlockedItems(filteredDataStreamsResult.unBlockedItems);
+              setBlockedItems(filteredDataStreamsResult.blockedItems);
+            })
+            .catch(() => {
+              // set unBlockedItems to all when filter is failed
+              setUnBlockedItems((selectedItems as DataStream[]).map((item) => item.name));
+            })
+            .finally(() => {
+              setLoading(false);
+            });
+          break;
+        case INDEX_OP_TARGET_TYPE.ALIAS:
+          setHint("Cache will be cleared for the following aliases.");
+          setBlockHint("Cache will not be cleared for the following aliases because one or more indexes are closed or blocked.");
+          filterBlockedItems(services, selectedItems as IAlias[], indexBlocksTypes, INDEX_OP_TARGET_TYPE.ALIAS, false)
+            .then((filteredAliasesResult) => {
+              setUnBlockedItems(filteredAliasesResult.unBlockedItems);
+              setBlockedItems(filteredAliasesResult.blockedItems);
+            })
+            .catch(() => {
+              // set unBlockedItems to all when filter is failed
+              setUnBlockedItems((selectedItems as IAlias[]).map((item) => item.alias));
+            })
+            .finally(() => {
+              setLoading(false);
+            });
+          break;
+        default:
+          setHint("Cache will be cleared for the following indexes.");
+          setBlockHint("Cache will not be cleared for the following indexes because they may be closed or blocked.");
+          filterBlockedItems(services, selectedItems as CatIndex[], indexBlocksTypes, INDEX_OP_TARGET_TYPE.INDEX, false)
+            .then((filteredIndexesResult) => {
+              setUnBlockedItems(filteredIndexesResult.unBlockedItems);
+              setBlockedItems(filteredIndexesResult.blockedItems);
+            })
+            .catch(() => {
+              // set unBlockedItems to all when filter is failed
+              setUnBlockedItems((selectedItems as CatIndex[]).map((item) => item.index));
+            })
+            .finally(() => {
+              setLoading(false);
+            });
+      }
+    } else if (!visible) {
+      setUnBlockedItems([] as string[]);
+      setBlockedItems([] as string[]);
+      setLoading(false);
```

Comment:
const [loading, setLoading] = useState(false); the initial value of loading is `false`, so I think we should reset the value to `false`.

Reply:
When visible is false, users change selectedItems, and this will set loading `false`. And if this time visible flip from `false` to `true`, the modal will come out for a little time?